### ISSUE TITLE
Subtitle incorrectly rendered with STKaiti font

### ISSUE THREAD

**Issue by 191919** [lhc70000/iina#436]
**System and IINA version:**

- OS X 10.11.6
- IINA 0.0.6

If I use STKaiti as my subtitle font, the display is wrong.

**Expected behavior:**

This is rendered by mpv.

![mpv-shot0001](https://cloud.githubusercontent.com/assets/298613/23242918/98288652-f9b6-11e6-9b0f-d5afef377e31.jpg)

**Actual behavior:**

This is by IINA.

![little forest natsu aki 2014 1080p bluray x264 dts-yukosu-0002](https://cloud.githubusercontent.com/assets/298613/23242919/98b70d00-f9b6-11e6-98f5-ca8f5aea7800.jpg)

**Steps to reproduce:**

Choose "STKaiti" in Subtitle pane of the preference window, then load any movie with .srt subtitle (to avoid customized fonts in some formats).

**How often does this happen?**

Always.

**Comment by lhc70000** [comment]
Please give me your subtitle file if possible.

**Comment by 191919** [comment]
The font (STKaiti) is included by default installations of OS X. You can check it in Font Book.app.

The physical path is "/Library/Fonts/Kaiti.ttc".

**Comment by lhc70000** [comment]
No, I'm always using STKaiti and didn't encountered this problem so far. It seems to be a encoding problem, so **your srt file is needed for testing**.

Did you installed mpv with uchardet?
Do other srt subtitles have the same problem?

**Comment by 191919** [comment]
The subtitle file is encoded with UTF-8. All of my subtitles reproduce the problem.

If I use IINA with any other font, the display is correct. STKaiti is the only font that causes the problem.

![little forest natsu aki 2014 1080p bluray x264 dts-yukosu-0001](https://cloud.githubusercontent.com/assets/298613/23300349/135ad4e2-fac0-11e6-8286-45e1393b2aba.jpg)
(STXingKai)

![little forest natsu aki 2014 1080p bluray x264 dts-yukosu-0002](https://cloud.githubusercontent.com/assets/298613/23300382/36fce340-fac0-11e6-8aec-389a4da45e07.jpg)
(STHeiti)

![little forest natsu aki 2014 1080p bluray x264 dts-yukosu-0003](https://cloud.githubusercontent.com/assets/298613/23300402/555e4040-fac0-11e6-9d41-9e429409eb63.jpg)
(Adobe Song Std)

**Comment by 191919** [comment]
The srt file is attached.

[Little.Forest.Natsu.Aki.2014.1080p.BluRay.x264.DTS-Yukosu.srt.zip](https://github.com/lhc70000/iina/files/799026/Little.Forest.Natsu.Aki.2014.1080p.BluRay.x264.DTS-Yukosu.srt.zip)

**Comment by megadr01d** [comment]
I found that a lot of fonts don't work. After selecting them in IINA's font panel they always revert to LucidaGrande.

This one works

```
[   1.658][v][sub/ass] fontselect: (DINAlternate-Bold, 400, 0) -> /Library/Fonts/DIN Alternate Bold.ttf, -1, DINAlternate-Bold
```

These don't

```
[   1.706][v][sub/ass] fontselect: (ArialRoundedMTBold, 400, 0) -> /System/Library/Fonts/LucidaGrande.ttc, -1, LucidaGrande
[   2.390][v][sub/ass] fontselect: (DINCondensed-Bold, 400, 0) -> /System/Library/Fonts/LucidaGrande.ttc, -1, LucidaGrande
```

**Comment by gkr41** [comment]
> I found that a lot of fonts don't work. After selecting them in IINA's font panel they always revert to LucidaGrande.

I have the same problem. Arial Unicode MS always falls back to Lucida Grande, even on embedded English subtitles.
Sorry for the reviving, but do we have any update on this?

- macOS 10.12.6
- IINA 0.0.14.1

**Comment by mrelwood** [comment]
Most fonts still fall back to default, including my favourites Skia, AppleSDGothicNeo and Trebuchet. Also: Arial, Helvetica Neue, Geeza, Geneva, Avenir, Marker Felt, Menlo, etc. Some fonts show as Lucida Grande and only change the spacing.

- macOS 10.13.2
- IINA 0.0.15 build 64

I also noticed a bug: After filtering (searching) the available fonts using any characters, I can no longer get it to show all fonts. If I filter with a character such as period, I get no results. Remove the period, still no results.

The font preview text should be a lot longer than just "Test". For example the phrase "A quick brown fox jumps over the lazy dog" that uses all English alphabets. I realized it can be edited, but the phrase above would be a lot better default as it is not clear that it can be edited.

**Comment by alejx** [comment]
@mrelwood 
> I also noticed a bug: After filtering (searching) the available fonts using any characters, I can no longer get it to show all fonts. If I filter with a character such as period, I get no results. Remove the period, still no results.

This bug should be fixed in the develop branch.

**Comment by yongtao-wang** [comment]
2024 and this issue is still there.

**Comment by missile-xuan** [comment]
> 2024 and this issue is still there.

好像是系统更新的问题  试试在通用语言中 加个英语试试
<img width="353" alt="image" src="https://github.com/user-attachments/assets/d02011b8-c7c4-4fb9-a27b-115d5c2e71e2">

https://bbs.tgfcer.com/thread-8392140-1-1.html

**Comment by proailurus** [comment]
I have the same issue.

macOS 15.1.1
IINA 1.3.5

mpv displays custom fonts correctly, both with `vo=gpu-next` and `vo=libmpv`.
